### DISCUSSION TITLE
EZSP 13 / EmberZnet 7.4.x firmware for Eachen/eWelink ZB-GW03 gateway

### DISCUSSION THREAD

**Discussion by PsychoRS** [Koenkk/zigbee2mqtt#22546] [Devices]
Hi,

Does anyone know where to get a 7.4.x firmware (EZSP 13) for the ZB-GW03 (sold as many brands, as eWelink/Eachen ZB Bridge Gateway.

This bridge consists in an ESP32 and a Silicon Labs EFR32MG21 Zigbee radio (CoolKit-Technologies "SM-011 V1.0" module). 

![ZY_ZigbeeGW](https://github.com/Koenkk/zigbee2mqtt/assets/552514/90d5d286-14c7-4350-8c67-88793b5d8a29)

Any chance to get EZSP 13 firmware for using the new ember driver?

**Comment by fuatakgun** [comment]
I have been testing this with EZSP 13 and it has been working well.

https://github.com/xsp1989/zigbeeFirmware/tree/master/firmware/ZigbeeBridge_SM-011-signed

**Reply by PsychoRS** [reply]
Is there any way to migrate from 6.7.8 and ezsp driver to 7.4.x and ember driver without having to re-pair all devices?

Thanks

**Reply by fuatakgun** [reply]
I have seen some examples about dumping the internal memory first and restoring it afterwards. I personally did not go into that hassle so started fresh :)

**Reply by Hedda** [reply]
While it is not an official check out this discssions which contains third-party backup and restore workaround which have been proven to work as a solution to avoid re-pairing all devices:

* https://github.com/Koenkk/zigbee2mqtt/discussions/16478

Also check out this guide:

* https://smarthomescene.com/guides/how-to-use-ember-driver-with-efr32mg21-zigbee-adapters/

Of course as well suggest searching for "migration" here to find other general Zigbee2MQTT (Z2M) discussions on migration topic:

* https://github.com/Koenkk/zigbee2mqtt/discussions

**Comment by PsychoRS** [comment]
Done and working like a charm. No need to repair any device.

1 - Update device zigbee firmware from 6.x.x to 7.4.3.0 (EZSP v13)
2 - Start zigbee2mqtt WITHOUT switching ezsp driver to ember driver, zigbee2mqtt will make a coordinator backup in EZSP v13 format, compatible with ember driver
4 - Stop Z2M, change driver to ember and launch again, ember driver will restore EZSP v13 backup, you may only have to power-cycle plugged devices or force an "update" on battery ones for them to sync again.

If you update zigbee firmware from 6.x.x.x to 7.4.3.0 and change ezsp driver to ember one in Z2M, all at once, you wouldn't have a valid coordinator backup (the ember driver needs a ezsp v12+ backup for restoring) and will loose all the devices.

**Comment by y8s** [comment]
for those of you running tasmota and wanting to run the ember driver, here's a process that worked great for me.

I was coming from the [thehelpfulidiot.com](https://thehelpfulidiot.com/a-wired-sonoff-zigbee-alternative) guide install at verison 13.0.0 on the eachen ewelink zb-gw03 and ezsp version 6.7.8 and it was working OK but that's not enough. I want more!

My proces follows this:
(a mostly copypasta hybrid of https://github.com/xsp1989/zigbeeFirmware/issues/49#issuecomment-2137189283 and the reply above mine https://github.com/Koenkk/zigbee2mqtt/discussions/22546#discussioncomment-10372934 )


1 - Make a ZB-GW03 Tasmota configuration backup
2 - update Tasmota to this: https://github.com/tasmota/install/blob/firmware/firmware/unofficial/tasmota32-zigbeebridge.bin (14.2.0.4 at the time I wrote this)
3 - backup tasmota configuration again (can't hurt and ensures everything is good)
4 - Reset Tasmota configuration (I chose command `reset 6` to preserve wifi and mqtt settings)
5 - Update device zigbee firmware from 6.x.x to 7.4.3.0 (EZSP v13) using the tasmota file upload box: https://github.com/xsp1989/zigbeeFirmware/blob/master/firmware/ZigbeeBridge_SM-011-signed/ncp-uart-sw_7.4.3.0_115200.ota 
6 - Restore the backup configuration file from step 3
7 - Stop and restart zigbee2mqtt WITHOUT switching `ezsp` driver to `ember` driver, zigbee2mqtt will make a coordinator backup in EZSP v13 format, compatible with ember driver
8 - Stop Z2M, change driver to `ember`
9 - Launch Z2M again, ember driver will restore EZSP v13 backup, you may only have to power-cycle plugged devices or force an "update" on battery ones for them to sync again.

Most devices just came right back up.

**Reply by kotborealis** [reply]
Just wanted to say - thank you so much! Only this guide worked for me.
The `reset 6` step is necessary - firmware upgrade won't work otherwise.
Did not have to repair any devices so far.

**Reply by y8s** [reply]
> Just wanted to say - thank you so much! Only this guide worked for me. The `reset 6` step is necessary - firmware upgrade won't work otherwise. Did not have to repair any devices so far.

Glad I could help. Sometimes it's just hard to find all the right information in one place so I put it all here.

**Comment by Hedda** [comment]
FYI, same questions and answers here -> https://github.com/arendst/Tasmota/discussions/21361

Again, newer firmware only works If you are using [ZHA](https://www.home-assistant.io/integrations/zha/) or [Zigbee2MQTT](https://www.zigbee2mqtt.io/) with Tasmota as [Serial-to-TCP-Bridge](https://tasmota.github.io/docs/Serial-to-TCP-Bridge/) and not if using  the Zigbee2Tasmota (a.k.a. Z2T) solution as a native Zigbee Gateway (which does not yet support 7.0.x firmware or later), see -> https://tasmota.github.io/docs/Zigbee/

**Comment by andrea-aj** [comment]
Hello,
I have a ZB GW03 V1.4 running Tasmota 13.2.0. The board includes a ZYZBP008 Zigbee adapter with an MG21 chip.

How can I check the firmware version of the Zigbee adapter?
Is it possible to update the adapter’s firmware using the information provided on this page?
Also, can Tasmota be upgraded to a newer version? If so, which version is recommended?

Thank you.
Andrea

**Comment by mvcorrea-ufc** [comment]
A question!
On the gw03 having tasmota and also having a EFR32MG21 on it. I love simple setups! Its possible to set the gw03 as a matter bridge (allowing the access my zigbee devices via my border router) like the sonoff zbbridge-u?
Thanks
